telemetry/src/lib.rs

```diff
@@ -46,13 +108,79 @@ impl TelemetryConfig {
         }
     }
 
-    pub fn try_init(self) -> Result<(), tracing_subscriber::util::TryInitError> {
+    /// Enable OpenTelemetry/OTLP tracing with the specified configuration.
+    #[must_use]
+    pub fn with_opentelemetry(self, config: OpenTelemetryConfig) -> Self {
+        Self {
+            otel: Some(config),
+            ..self
+        }
+    }
+
+    /// Initialize the OpenTelemetry TracerProvider and set it globally.
+    fn init_opentelemetry(&self)
+                          -> Result<(trace::TracerProvider, trace::Tracer), Box<dyn std::error::Error>>
+    {
+        // Build an OpenTelemetry Resource with service metadata
+        let resource = Resource::new(vec![
+            KeyValue::new("service.name", self.service_name.clone()),
+            KeyValue::new("service.version", self.service_version.clone()),
+            KeyValue::new("deployment.environment", self.environment.clone()),
+        ]);
+
+        let config = self.otel.as_ref().expect("OpenTelemetry config must be present");
+
+        let exporter = opentelemetry_otlp::new_exporter()
+            .tonic()
+            .with_endpoint(&config.endpoint)
+            .build_span_exporter()?;
+
+        let trace_config = trace::config()
+            .with_resource(resource)
+            .with_sampler(Sampler::AlwaysOn);
+
+        // Create a new tracer provider builder
+        let tracer_provider = trace::TracerProvider::builder()
+            .with_config(trace_config)
+            .with_batch_exporter(exporter, Tokio)
+            .build();
+
+        // Create a concrete tracer from the provider:
+        let tracer = tracer_provider.tracer("telemetry");
+
+        // Now set the global tracer provider (if desired)
+        global::set_tracer_provider(tracer_provider.clone());
+        global::set_text_map_propagator(TraceContextPropagator::new());
+
+        Ok((tracer_provider, tracer))
+    }
+
+
+    /// Try to initialize telemetry (journald/stderr + optional OTLP).
+    /// Returns an error if something goes wrong setting up the subscriber stack.
+    pub fn try_init(self) -> Result<(TelemetryShutdownHandler, Result<(), tracing_subscriber::util::TryInitError>), Box<dyn std::error::Error>> {
```

Comment:
Ths type is quite strange. I think you can simplify it by using `thiserror` for your error type, and the type signature then becomes:

```rust
pub fn try_init(self) -> Result<TelemetryShutdownHandler, OrbTelemetryErr> {
    // ...
}
```